### PR TITLE
add persistentImages

### PR DESCRIPTION
**Pull request by T0MASD**
Add persistentImages feature which perserves local images between ocm-container sessions

### CODE REVIEW

**Comment by T0MASD** [comment]
/hold for fix to left over containers
```
$ podman ps -aq
bd2a8b53fd5c
e8bfe9d90987
```

File: pkg/featureSet/persistentImages/persistentImages.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	config := &Config{}
+
+	ocmClient, err := ocm.NewClient()
+	if err != nil {
+		return config, err
+	}
+	defer ocmClient.Close()
+
+	mount := filepath.Join(home, sourceDir)
```
Comment (clcollins) [inline pkg/featureSet/persistentImages/persistentImages.go:32]:
```suggestion
	log.Debug("using image cache mount: " + mount)
	mount := filepath.Join(home, sourceDir)
```

Not a deal breaker at all - just suggesting some debug logging for troubleshooting.

File: pkg/featureSet/persistentImages/persistentImages.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package persistentImages
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/ocm-container/pkg/engine"
```
Comment (clcollins) [inline pkg/featureSet/persistentImages/persistentImages.go:7]:
```suggestion
	"github.com/openshift/ocm-container/pkg/engine"
	log "github.com/sirupsen/logrus"
```

Related to the debug log comment below...

**Review by clcollins — CHANGES_REQUESTED** [review]
This looks great!  Other than the ocm client comment and a suggestion for a debug log line, I'm good with this.

I see you have this held related to container cleanup?  What is the issue you're seeing there?

File: pkg/featureSet/persistentImages/persistentImages.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	Env    map[string]string
+	Mounts []engine.VolumeMount
+}
+
+func New(home string) (*Config, error) {
+	var err error
+
+	config := &Config{}
+
+	ocmClient, err := ocm.NewClient()
```
[the remark this answers is not in the document]
Comment (T0MASD) [reply pkg/featureSet/persistentImages/persistentImages.go]: good catch, it's from copy/pasta, removed in 191364e55b09332620264c71b3ba1387fa44f6ba

**Pull request merged**